share/spice/islamic_prayer_times/islamic_prayer_times.js

```diff
@@ -0,0 +1,98 @@
+(function (env) {
+    "use strict";
+
+    function setDst(time, dst) {
+        return dst === 1 ? time.add(dst, 'hours') : time;
+    }
+
+    function tz(time, offset) {
+        var clone = time.clone();
+        clone.utcOffset(offset);
+        clone.add(time.utcOffset() - clone.utcOffset(), 'minutes');
+        return clone;
+    }
+
+    function getClosest(timings, offset, local) {
```

Comment:
I think this code warrant some comments to explain what these are for and why we need them.